src/resolve-info.ts

```diff
@@ -59,11 +74,27 @@ export interface TypeExpansion {
   [fieldName: string]: FieldExpansion | LeafField;
 }
 
-function createLeafField<T extends object>(props: T): T & LeafField {
-  return {
-    [LeafFieldSymbol]: true,
-    ...props
-  };
+function createLeafField<T extends object>(
+  props: T,
+  shouldInclude: (variables: ShouldIncludeVariables) => boolean
```

Comment:
The typename "ShouldIncludeVariables" should be `ShouldIncludeParams`. Otherwise it would read `variables.variables[variableName]`.

Reply:
renamed.